Feature/make angular15 compatible

dsteegen opened the pull request:
This PR will make the library compatible with Angular 15. The changes are inspired by this [fork](https://github.com/fast-facts/ngx-bootstrap-multiselect).

Commit:
Made the library compatible with Angular 15 based on https://github.com/fast-facts/ngx-bootstrap-multiselect

Commit:
Ignore .angular folder.

Commit:
Added demo project to illustrate the usage of the library.

Commit:
Update readme so it is clear the library is now compiled with Ivy iso the View Engine.

Commit:
Reformat tables in README file.

sommmen commented:
Shouldn't the deps like 

```
@angular/core": "15.2.0
```

Be ^15.0.0?

Commit:
Added Licence file.

Commit:
Updated Angular dependency version to ^15.0.0 iso setting it to a fixed version.

dsteegen commented:
> Shouldn't the deps like
> 
> ```
> @angular/core": "15.2.0
> ```
> 
> Be ^15.0.0?

Yes! That is indeed better than fixating it to 15.2.0. I already updated it in this PR.

Most changes were taken over from another fork so if you spot other things that can be improved; let me know :)

sommmen commented:
> > Shouldn't the deps like
> > ```
> > @angular/core": "15.2.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Be ^15.0.0?
> 
> Yes! That is indeed better than fixating it to 15.2.0. I already updated it in this PR.
> 
> Most changes were taken over from another fork so if you spot other things that can be improved; let me know :)

I'll take a look tomorrow, am on phone atm.

dsteegen commented:
> > 
> 
> Uh also, I'm not a js dev, is it easy to create a beta package I can test with my app? We kind of use all features extensively so testing the pr would be nice for me.

You can build it manually if you clone my repository and switch to my branch. Afterwards you have to execute the following steps:

- npm install in the root folder
- npm run build
- in the dist folder, run npm pack. This should create ngx-bootstrap-multiselect-3.0.0.tgz inside the dist folder
- in your project's package.json, replace the ngx-bootstrap-multiselect entry with `"ngx-bootstrap-multiselect": "/Users/davysteegen/ngx-bootstrap-multiselect/dist/ngx-bootstrap-multiselect-3.0.0.tgz",` (change the directory to match your own)
- run npm install on your project's package.json.

You should now have the version that corresponds to this PR.

dsteegen commented:
@sommmen 
FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated

The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.

sommmen commented on package.json:
```diff
@@ -12,12 +22,31 @@
     "multi select"
   ],
   "license": "MIT",
-  "peerDependencies": {
```
Seems you removed the peer dependencies - is that o.k?
The angular packages are defined as dev deps - but i can imagine you'd need those 3 packages at runtime.

again, no js dev so not sure.

dsteegen replied:
That's a valid comment! It's indeed better to specify the peer dependencies to make sure that the project using this library is compatible with the specified dependencies. I will try to add those this evening or tomorrow :)

sommmen reviewed (commented):
LGTM!

sommmen commented:
> @sommmen FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated
> 
> The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.

Hiya - thanks for sharing your knowledge.
I forget i'm still on Angular 14, so i'd have to upgrade Angular in the future.

dsteegen commented:
> > @sommmen FYI: if you don't want to build a local version yourself, your can always use a published version from https://www.npmjs.com/package/ngx-bootstrap-multiselect-updated
> > The fork/PR I created is based on that NPM package (which is also a fork from the original ngx-bootstrap-multiselect from @softsimon). I hope this helps.
> 
> Hiya - thanks for sharing your knowledge. I forget i'm still on Angular 14, so i'd have to upgrade Angular in the future.

Normally you would still be able to use this library with Angular 14 since there are no breaking API changes compared to version 15.

Commit:
- Updated structure according to angular-cli generates it.
- angular.json updated to have a build for the library and a build for serving the demo.
- Added peerDependencies for the library

sommmen commented:
@softsimon don't forget to create a release ❤️

dsteegen commented:
@softsimon if there is anything we do from our end to smoothen the release, just let us know. If not, I appreciate the time and effort you are putting into this 🙂

softsimon commented:
@sommmen 
@dsteegen 

```
npm notice Publishing to https://registry.npmjs.org/
+ ngx-bootstrap-multiselect@3.0.0
```

dsteegen commented:
Cheers! Thank you for taking care of this.
